gems/sorbet-runtime/test/types/method_validation.rb

```diff
@@ -306,6 +306,158 @@ def @mod.foo
         assert_empty(lines[3..-1])
       end
 
+      describe 'ranges' do
+        describe 'return type is non-nilable integer' do
+          it 'permits a range that has integers on start and end' do
+            @mod.sig { returns(T::Range[Integer] )}
+            def @mod.foo
+              (1...10)
+            end
+
+            assert_equal((1...10), @mod.foo )
+          end
+
+          it 'permits a range that has an integer start and no end' do
+            @mod.sig { returns(T::Range[Integer] )}
+            def @mod.foo
+              (1...nil)
+            end
+
+            assert_equal((1...nil), @mod.foo )
+          end
+
+          # Ruby 2.6 does not support ranges with boundless starts
+          if RUBY_VERSION >= '2.7'
+            it 'permits a range that has an integer start and no end' do
+              @mod.sig { returns(T::Range[Integer] )}
+              def @mod.foo
+                (nil...10)
+              end
+
+              assert_equal((nil...10), @mod.foo )
+            end
+          end
+
+          it 'permits a range with no beginning or end' do
+            @mod.sig { returns(T::Range[Integer] )}
+            def @mod.foo
+              (nil...nil)
+            end
+
+            assert_equal((nil...nil), @mod.foo )
+          end
+        end
+
+        describe 'return type is nilable integer' do
+          it 'permits a range that has integers on start and end' do
+            @mod.sig { returns(T::Range[T.nilable(Integer)] )}
+            def @mod.foo
+              (1...10)
+            end
+
+            assert_equal((1...10), @mod.foo )
+          end
+
+          it 'permits a range that has an integer start and no end' do
+            @mod.sig { returns(T::Range[T.nilable(Integer)] )}
+            def @mod.foo
+              (1...nil)
+            end
+
+            assert_equal((1...nil), @mod.foo )
+          end
+
+          # Ruby 2.6 does not support ranges with boundless starts
+          if RUBY_VERSION >= '2.7'
+            it 'permits a range that has an integer start and no end' do
+              @mod.sig { returns(T::Range[T.nilable(Integer)] )}
+              def @mod.foo
+                (nil...10)
+              end
+
+              assert_equal((nil...10), @mod.foo )
+            end
+          end
+
+          it 'rejects a range with no start and no end' do
+            @mod.sig { returns(T::Range[T.nilable(Integer)] )}
+            def @mod.foo
+              (nil...nil)
+            end
+
+            assert_equal((nil...nil), @mod.foo )
+          end
+        end
+
+        describe 'return type is a range of nils' do
```

Comment:
Similar to my comment above, these tests are only testing that `T::Range[Integer]` cannot be cast to `T::Range[NilClass]` (or that `T::Range[T.untyped]` can be cast to `T::Range[NilClass]`, for the final case) and don't convey more information about how Range values are being interpreted. I say we drop these tests.

Reply:
Yep makes sense!